tests/unit/tf/blocks/test_dlrm.py

```diff
@@ -16,77 +16,105 @@
+        mm.DLRMBlock(
+            schema, embedding_dim=64, bottom_block=mm.MLPBlock([64]), top_block=mm.MLPBlock([16])
         )
     assert "DLRM requires categorical features" in str(excinfo.value)
 
 
 def test_dlrm_block_single_categ_feature(testing_data: Dataset):
     schema = testing_data.schema.select_by_tag([Tags.ITEM_ID])
-    dlrm = ml.DLRMBlock(schema, embedding_dim=64, top_block=ml.MLPBlock([32]))
-    outputs = dlrm(ml.sample_batch(testing_data, batch_size=100, include_targets=False))
+    dlrm = mm.DLRMBlock(schema, embedding_dim=64, top_block=mm.MLPBlock([32]))
+    outputs = dlrm(mm.sample_batch(testing_data, batch_size=100, include_targets=False))
 
     assert list(outputs.shape) == [100, 32]
 
 
 def test_dlrm_block_no_schema():
     with pytest.raises(ValueError) as excinfo:
-        ml.DLRMBlock(
+        mm.DLRMBlock(
             schema=None,
             embedding_dim=64,
-            bottom_block=ml.MLPBlock([64]),
-            top_block=ml.MLPBlock([32]),
+            bottom_block=mm.MLPBlock([64]),
+            top_block=mm.MLPBlock([32]),
         )
     assert "The schema is required by DLRM" in str(excinfo.value)
 
 
 def test_dlrm_block_no_bottom_block(testing_data: Dataset):
     with pytest.raises(ValueError) as excinfo:
-        ml.DLRMBlock(schema=testing_data.schema, embedding_dim=64, bottom_block=None)
+        mm.DLRMBlock(schema=testing_data.schema, embedding_dim=64, bottom_block=None)
     assert "The bottom_block is required by DLRM" in str(excinfo.value)
 
 
 def test_dlrm_emb_dim_do_not_match_bottom_mlp(testing_data: Dataset):
     with pytest.raises(ValueError) as excinfo:
-        ml.DLRMBlock(schema=testing_data.schema, bottom_block=ml.MLPBlock([64]), embedding_dim=75)
+        mm.DLRMBlock(schema=testing_data.schema, bottom_block=mm.MLPBlock([64]), embedding_dim=75)
     assert "needs to match the last layer of bottom MLP" in str(excinfo.value)
+
+
+def test_dlrm_raises_with_embeddings_and_options(testing_data: Dataset):
+    schema = testing_data.schema
+    with pytest.raises(ValueError) as excinfo:
+        mm.DLRMBlock(
+            schema,
+            embedding_dim=10,
+            embedding_options=mm.EmbeddingOptions(),
+            embeddings=mm.Embeddings(schema),
+        )
+    assert "Only one-of `embeddings` or `embedding_options` may be provided" in str(excinfo.value)
+
+
+def test_dlrm_with_embeddings(testing_data: Dataset):
+    schema = testing_data.schema
+    dim = 12
+    dlrm = mm.DLRMBlock(
+        schema,
+        embeddings=mm.SequentialBlock(
+            mm.AsRaggedFeatures(),
+            mm.Embeddings(schema, infer_embedding_sizes=False, embedding_dim_default=dim),
+        ),
+        bottom_block=mm.MLPBlock([dim]),
+    )
+    outputs = dlrm(mm.sample_batch(testing_data, batch_size=100, include_targets=False))
+
+    assert list(outputs.shape) == [100, (dim * dim - dim) / 2]
```

Comment:
It's not clear to me if this output dimension is correct based on the description of the `DotProductInteraction`

https://github.com/NVIDIA-Merlin/models/blob/6d47dd3872a1b3f332a76c483dcf8d3155429ef0/merlin/models/tf/blocks/interaction.py#L31-L33

Reply:
Added a PR to change the stacking here assuming my understanding is correct here. https://github.com/NVIDIA-Merlin/models/pull/637